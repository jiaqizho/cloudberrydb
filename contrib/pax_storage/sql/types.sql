
CREATE TABLE pax_test.all_typbyval_pg_types (
  id int,
  bool_col bool,
  char_col char,
  int2_col int2,
  cid_col cid,
  
  float4_col float4,
  int4_col int4,
  date_col date,
  
  oid_col oid,
  -- Ã¥xid_col xid,
  time_stamp_col timestamp,
  int8_col int8,
  -- xid8_col xid8,
  float8_col float8,
  money_col money,
  time_col time,
  timestamptz_col timestamptz,
  pg_lsn_col pg_lsn
) USING pax distributed by (id);

insert into pax_test.all_typbyval_pg_types values(1, true,'c',2,'cid',4.2,5,'2023-05-17 17:56:49',7,'2023-05-17 17:56:49',10,11.1111,12,'2023-05-17 17:56:49','2023-05-17 17:56:49', '16/0'),
(1, true,'c',2,'cid',4.2,5,'2023-05-17 17:56:49',7,'2023-05-17 17:56:49',10,11.1111,12,'2023-05-17 17:56:49','2023-05-17 17:56:49', '16/0'),
(1, true,'c',2,'cid',4.2,5,'2023-05-17 17:56:49',7,'2023-05-17 17:56:49',10,11.1111,12,'2023-05-17 17:56:49','2023-05-17 17:56:49', '16/0');
select * from pax_test.all_typbyval_pg_types;

create table pax_test.all_typlen_lt_0_pg_type (
  id int,
  name_col name,
  numeric_col numeric,
  text_col text,
  varchar_col varchar(128),
  point_col point
) USING pax distributed by (id);

insert into pax_test.all_typlen_lt_0_pg_type values(1,'hello', 1.23, 'text', 'varchar', point(1,2));
select * from pax_test.all_typlen_lt_0_pg_type;

drop table pax_test.all_typbyval_pg_types;
drop table pax_test.all_typlen_lt_0_pg_type;
