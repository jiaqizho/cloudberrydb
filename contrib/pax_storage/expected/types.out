set default_table_access_method = pax;
CREATE TABLE pax_test.all_typbyval_pg_types (
  id int,
  bool_col bool,
  char_col char,
  int2_col int2,
  cid_col cid,
  
  float4_col float4,
  int4_col int4,
  date_col date,
  
  oid_col oid,
  -- Ã¥xid_col xid,
  time_stamp_col timestamp,
  int8_col int8,
  -- xid8_col xid8,
  float8_col float8,
  money_col money,
  time_col time,
  timestamptz_col timestamptz,
  pg_lsn_col pg_lsn
) USING pax distributed by (id);
insert into pax_test.all_typbyval_pg_types values(1, true,'c',2,'cid',4.2,5,'2023-05-17 17:56:49',7,'2023-05-17 17:56:49',10,11.1111,12,'2023-05-17 17:56:49','2023-05-17 17:56:49', '16/0'),
(1, true,'c',2,'cid',4.2,5,'2023-05-17 17:56:49',7,'2023-05-17 17:56:49',10,11.1111,12,'2023-05-17 17:56:49','2023-05-17 17:56:49', '16/0'),
(1, true,'c',2,'cid',4.2,5,'2023-05-17 17:56:49',7,'2023-05-17 17:56:49',10,11.1111,12,'2023-05-17 17:56:49','2023-05-17 17:56:49', '16/0');
select * from pax_test.all_typbyval_pg_types;
 id | bool_col | char_col | int2_col | cid_col | float4_col | int4_col |  date_col  | oid_col |      time_stamp_col      | int8_col | float8_col | money_col | time_col |       timestamptz_col        | pg_lsn_col 
----+----------+----------+----------+---------+------------+----------+------------+---------+--------------------------+----------+------------+-----------+----------+------------------------------+------------
  1 | t        | c        |        2 |       0 |        4.2 |        5 | 05-17-2023 |       7 | Wed May 17 17:56:49 2023 |       10 |    11.1111 |    $12.00 | 17:56:49 | Wed May 17 17:56:49 2023 PDT | 16/0
  1 | t        | c        |        2 |       0 |        4.2 |        5 | 05-17-2023 |       7 | Wed May 17 17:56:49 2023 |       10 |    11.1111 |    $12.00 | 17:56:49 | Wed May 17 17:56:49 2023 PDT | 16/0
  1 | t        | c        |        2 |       0 |        4.2 |        5 | 05-17-2023 |       7 | Wed May 17 17:56:49 2023 |       10 |    11.1111 |    $12.00 | 17:56:49 | Wed May 17 17:56:49 2023 PDT | 16/0
(3 rows)

create table pax_test.all_typlen_lt_0_pg_type (
  id int,
  name_col name,
  numeric_col numeric,
  text_col text,
  varchar_col varchar(128),
  point_col point
) USING pax distributed by (id);
insert into pax_test.all_typlen_lt_0_pg_type values(1,'hello', 1.23, 'text', 'varchar', point(1,2));
select * from pax_test.all_typlen_lt_0_pg_type;
 id | name_col | numeric_col | text_col | varchar_col | point_col 
----+----------+-------------+----------+-------------+-----------
  1 | hello    |        1.23 | text     | varchar     | (1,2)
(1 row)

drop table pax_test.all_typbyval_pg_types;
drop table pax_test.all_typlen_lt_0_pg_type;
